src/client/event-stream.ts

```diff
@@ -1,23 +1,71 @@
+import { ReReadable } from "@gameye/streamkit";
+import { OutgoingHttpHeaders } from "http";
 import { second } from "msecs";
 import * as querystring from "querystring";
 import { pipeline, Readable } from "stream";
 import { EndStream, FromJSONTransform, SplitTransform } from "../streams";
-import { createRequestStream, getResponse } from "../utils";
+import { createRequestStream, getResponse, retry, RetryConfig } from "../utils";
+import { Action } from "./action";
 
-export async function createHttpEventStream(
+export interface EventStreamRequestRetryConfig {
+    requestOptions?: EventStreamRequestConfig;
```

Comment:
prefer a flat options object. no nested options.

this will make more ergnomic api! we can do things like
```
EventStreamRequestRetryConfig(..., ... {heartbeatInterval:10000, retryCount:50})

```